### PR TITLE
Add simple export error handling

### LINKED ISSUE #667: Fix 422 errors incorrectly reported as 500
Sentry error: https://sentry.io/nusmods/export/issues/428820084/

Stack trace: 

```
TypeError: Cannot read property 'timetable' of undefined
  File "/home/ubuntu/nusmods-export/render.js", line 64, in injectData
    const moduleCodes = Object.keys(data.timetable);
  File "/home/ubuntu/nusmods-export/render.js", line 85, in Object.image
    const boundingBox = await injectData(page, data);

TypeError: Cannot read property 'timetable' of undefined
```

### PR DESCRIPTION
**Pull request by ZhangYiJiang**
Fixes #667 

This PR also updates the README file with more details, and adds some basic error pages so the user doesn't just see a string if something goes boom

### CODE REVIEW

File: export/README.md [hunk 1]
```diff
@@ -26,15 +26,28 @@ cd ../export
...
 - Dark mode 
 - Timetable orientation 
-- Hidden modules    
+- Hidden modules
+- Custom coloring 
+
+## Process 
+
+1. The user's client extracts the relevant slice of the Redux state necessary to render the timetable and serializes it into JSON  
+2. The serialized state is added as query to the 'Download as' link the user sees when they open the dropdown menu
```
Comment (yangshun) [inline export/README.md]: More maintainable to use a list of `1.`

Comment (yangshun) [inline export/README.md]: `added as a query param`

File: export/README.md [hunk 2]
```diff
@@ -26,15 +26,28 @@ cd ../export
...
+2. The serialized state is added as query to the 'Download as' link the user sees when they open the dropdown menu 
+3. When the user clicks on the link, the browser makes a GET request to the chosen export endpoint
+4. The endpoint parses and validates the incoming data  
+5. The export service uses the Puppeteer instance to create a new `Page` object
+6. The page loads a special version of the app that only has the `<TimetableContent>` component, and injects the data into the Redux store 
+7. The screenshot or PDF is taken of the page, and sent back to the user 
+
+## API
+
+Both export API endpoints accept GET requests and take a `data` query that consists of JSON encoded data in the shape of `ExportData`.
```
Comment (yangshun) [inline export/README.md]: query param

File: export/README.md [hunk 3]
```diff
@@ -44,4 +57,12 @@ Download PDF of the timetable.
 
 Download PNG image of the timetable. 
 
+#### Parameters 
+
+- `pixelRatio` (Number: 1 to 3 inclusive) - the device pixel ratio as reported in [`window.devicePixelRatio`](https://developer.mozilla.org/en-US/docs/Web/API/Window/devicePixelRatio). This scales the entire image by that amount
```
Comment (yangshun) [inline export/README.md]: Should `data` also be mentioned here?

File: export/views/500.pug [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  p
+    | Something went wrong. We've logged the error and will be looking into
+    | it.
+
+  if eventId
+    script(src='https://cdn.ravenjs.com/3.21.0/raven.min.js' crossorigin='anonymous')
+    script.
+      Raven.showReportDialog({
+        eventId: '#{eventId}',
+        dsn: 'https://37215374a3fd42a198c7b144c8886335@sentry.io/265295'
```
Comment (yangshun) [inline export/views/500.pug]: Is there a `config.js` to dump all these?

File: export/README.md [hunk 5]
```diff
@@ -21,27 +22,53 @@ yarn start
 yarn start:export
 
 # Start export server - use "yarn devtools" if need to see the graphical browser with  
-# developer tools. Note that PDF export does not work if this is set. 
+# developer tools. Note that PDF export in devtools mode.
```
Comment (taneliang) [inline export/README.md]: The second sentence is incomplete

**Pull request merged**